src/client/gui/lib/vm_table/zones_dropdown_button.dart

```diff
@@ -0,0 +1,124 @@
+            child: SizedBox(
+              width: 20,
+              height: 20,
+              child: CircularProgressIndicator(),
+            ),
+          ),
+        ],
+        error: (_, __) => [
+          const PopupMenuItem(
+            enabled: false,
+            child: Text('Error loading zones'),
+          ),
+        ],
+        data: (zones) => [
+          const PopupMenuItem(
+            enabled: false,
+            child: Padding(
+              padding: EdgeInsets.symmetric(horizontal: 4, vertical: 8),
+              child: Text(
+                'Availability zones',
+                style: TextStyle(
+                  fontSize: 24,
+                  fontWeight: FontWeight.w600,
+                  color: Colors.black,
+                ),
+              ),
+            ),
+          ),
+          ...zones.map((zone) {
+            // Count running instances in this zone
+            final runningCount = vmInfos
+                .where((vm) =>
+                    vm.zone?.name == zone.name &&
+                    vm.instanceStatus.status == Status.RUNNING)
+                .length;
+
+            return PopupMenuItem(
+              enabled: false,
+              padding: EdgeInsets.zero,
+              child: Container(
+                decoration: BoxDecoration(
+                  border: Border.all(color: Colors.grey.withOpacity(0.3)),
+                  borderRadius: BorderRadius.circular(4),
+                ),
+                margin: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
+                padding: const EdgeInsets.all(8),
+                child: _ZoneToggleRow(
+                  zone.name,
+                  zone.available,
+                  runningCount,
+                ),
+              ),
+            );
+          }).toList(),
+        ],
+      ),
+    );
+  }
+}
+
+class _ZoneToggleRow extends ConsumerWidget {
+  final String zoneName;
+  final bool available;
+  final int instanceCount;
+
+  const _ZoneToggleRow(this.zoneName, this.available, this.instanceCount);
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final client = ref.watch(grpcClientProvider);
+
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+      children: [
+        Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Text('$zoneName',
+                style: TextStyle(
+                  fontSize: 16,
+                  fontWeight: FontWeight.w600,
+                  color: Colors.black,
+                )),
+            Text(
+              '$instanceCount running instance${instanceCount == 1 ? '' : 's'}',
+              style: TextStyle(
+                fontSize: 12,
+                color: Colors.black,
+              ),
+            ),
+          ],
+        ),
+        Switch(
+          value: available,
+          onChanged: (value) => client.zonesState([zoneName], value),
+        ),
```

Comment:
Isn't this still the Flutter Switch instead of our own Switch from `switch.dart`?